backends/p4tools/modules/testgen/targets/bmv2/program_info.cpp

```diff
@@ -57,6 +57,7 @@ BMv2_V1ModelProgramInfo::BMv2_V1ModelProgramInfo(
     /// the deparser. This sequence also includes nodes that handle transitions between the
     /// individual component instantiations.
     int pipeIdx = 0;
+    pipelineSequence.emplace_back(Continuation::Guard(getPortConstraint(getTargetInputPortVar())));
```

Comment:
Do not use a guard here. Just a normal comparison for the ingress port variable. It needs to be smaller than 7.  Add this to the target constraints in the program info.